packages/utils/src/promisebuffer.ts

```diff
@@ -18,13 +21,22 @@ export class PromiseBuffer<T> {
   /**
    * Add a promise to the queue.
    *
-   * @param task Can be any PromiseLike<T>
+   * @param taskProducer A function producing any PromiseLike<T>
```

Comment:
Let’s make it clear in the doc string that we migrated from task -> taskProducer.

As extra context, can we also include why this change was made as a docstring comment? Just so it’s easier when we try to go grep for these changes in the future.

Reply:
```suggestion
   * @param taskProducer A function producing any PromiseLike<T>; In previous versions is used to be `@param task: PromiseLike<T>`, however, Promises were instantly created on the call-site, making them fell through the buffer limit.
```